witness/internal/witness/db.go

```diff
@@ -38,16 +38,24 @@ func (d *Database) init() error {
 	return err
 }
 
+type querier interface {
+	QueryRow(query string, args ...interface{}) *sql.Row
+}
+
 // GetLatest reads the latest checkpoint written to the DB for a given log.
 func (d *Database) GetLatest(logPK string) (*Chkpt, error) {
+	return d.getLatestChkpt(d.db, logPK)
+}
+
+func (d *Database) getLatestChkpt(q querier, logPK string) (*Chkpt, error) {
```

Comment:
```suggestion
func getLatestChkpt(q func(query string, args ...interface{}) *sql.Row, logPK string) (*Chkpt, error) {
```

a) drop this from being attached to `Database` as it isn't using any of the fields from it that I can see.
b) take the query function directly

You would then change `q.QueryRow(...)` below for just `q(...)` (so consider renaming the param `q` to be `queryRow`.

Callers would be changed along the lines of `d.getLatestChkpt(d.db,...)` to `d.getLatestChkpt(d.db.QueryRow, ...)`.

Reply:
Done, thank you!